src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -838,5 +842,263 @@ public static TransportSecurityBindingElement CreateSspiNegotiationOverTransport
+                        return;
+                    }
+
+                    ExportTransportSecurityBindingElement(dummyTransportBindingElement, transportTokenAssertionProvider, exporter, context);
+                }
+
+                hasCompletedSuccessfully = true;
+            }
+            finally
+            {
+                try
+                {
+                    exporter.State.Remove(SecurityPolicyStrings.SecureConversationBootstrapBindingElementsBelowSecurityKey);
+                }
+                catch (Exception e)
+                {
+                    // Always immediately rethrow fatal exceptions.
+                    if (hasCompletedSuccessfully || Fx.IsFatal(e)) throw;
+                }
+            }
+        }
+
+        //
+        // We will emit the wssp trust 10 assertion for all the case except for the basic http binding
+        // created through the BasicHttpBinding class.  The reason for this exception is to allow better 
+        // interop with third party when the third party doesn't understand the trust asserion
+        //
+        private static bool RequiresWsspTrust(SecurityBindingElement sbe)
+        {
+            if (sbe == null)
+                return false;
+
+            return !sbe.DoNotEmitTrust;
+        }
+
+        public static void ExportTransportSecurityBindingElement(TransportSecurityBindingElement binding, ITransportTokenAssertionProvider transportTokenAssertionProvider, MetadataExporter exporter, PolicyConversionContext policyContext)
+        {
+            WSSecurityPolicy sp = WSSecurityPolicy.GetSecurityPolicyDriver(binding.MessageSecurityVersion);
+
+            if (transportTokenAssertionProvider == null && binding.AllowInsecureTransport)
+            {
+                if ((policyContext != null) && (policyContext.BindingElements != null))
+                {
+                    foreach (BindingElement be in policyContext.BindingElements)
+                    {
+                        if (be.GetType().FullName.Equals("CoreWCF.Channels.HttpTransportBindingElement"))
+                        {
+                            throw new Exception("Work out how to do this some other way");
+                            //transportTokenAssertionProvider = new HttpsTransportBindingElement();
+                            //break;
+                        }
+
+                        if (be.GetType().FullName.Equals("CoreWCF.Channels.TcpTransportBindingElement"))
+                        {
+                            transportTokenAssertionProvider = new SslStreamSecurityBindingElement();
+                            break;
+                        }
+                    }
+                }
+            }
+
+            XmlElement transportTokenAssertion = transportTokenAssertionProvider.GetTransportTokenAssertion();
+
+            if (transportTokenAssertion == null)
+            {
+                if (transportTokenAssertion == null)
```

Comment:
Thanks. This was left over from where some other code ran to try and generate the transportTokenAssertion, and if it was still null, then it would throw. I refactored to have the caller always generate it and didn't clean this up correctly.